docs: add additional policy document (#3755)
aws/copilot-cli: site/content/docs/manifest/pipeline.en.md

<span class="parent-field">build.</span><a id="build-buildspec" href="#build-buildspec" class="field">`buildspec`</a> <span class="type">String</span>
Optional. The URI that identifies a buildspec to use for this build project. By default, Copilot will generate one for you, located at `copilot/pipelines/[your pipeline name]/buildspec.yml`.

> <span class="parent-field">build.</span><a id="additional-policy" href="#additional-policy" class="field">`additional_policy.`</a><a id="policy-document" href="#policy-document" class="field">`PolicyDocument`</a> <span class="type">Map</span>

Reviewer: ```suggestion
<span class="parent-field">build.</span><a id="build-additional-policy" href="#build-additional-policy" class="field">`additional_policy.`</a><a id="policy-document" href="#policy-document" class="field">`PolicyDocument`</a> <span class="type">Map</span>
```

Reviewer 2: Should we just stop at `build.additional_policy` instead of `build.additional_poilcy.PolicyDocument`?
It seems like `PolicyDocument` is not a manifest field but a part of the policy document (i.e. a part of the input to `additional_policy`

Author: I added `PolicyDocument` as well in order for customers to understand that they need to specify `PolicyDocument` keyword followed by the document body. Wouldn't it be confusing for the customers to see only `additional_policy` keyword and then suddenly see `PolicyDocument` in the example code?

> Optional. Specify additional policy document to upgrade build project role.

Reviewer: ```suggestion
Optional. Specify an additional policy document to add to the build project role.
```

> The additional policy document can be specified in JSON, for example:

Reviewer: i think maybe do the yaml example first? and say alternatively in json?

Reviewer: also maybe have the example policy document be the same between the two?

> The additional policy document can be specified in a map in YAML, for example:

Reviewer 3: Do we have plans to allow pointing to a file?

Author: Sorry I am not sure which file you are talking about? Do you mean manifest file?

```yaml
build:
  additional_policy:
    PolicyDocument:
      Version: 2012-10-17
      Statement:
        - Effect: Allow
          Action:
            - ecr:GetAuthorizationToken
          Resource: '*'
```
or alternatively as JSON:

> ```json
> build:
>   additional_policy:
>     PolicyDocument: {“Statement":[{"Action":["ecr:GetAuthorizationToken"],"Effect":"Allow","Resource":"*"}],"Version":"2012-10-17"}
> ```

Reviewer 3: ```suggestion
```yaml
build:
  additional_policy:
    PolicyDocument: >
      {
        "Statement": [
          {
            "Action": ["ecr:GetAuthorizationToken"],
            "Effect": "Allow",
            "Resource": "*"
          }
        ],
        "Version":"2012-10-17"
      }
 ```
